src/templates/edit.tsx

```diff
@@ -58,6 +58,24 @@ const Edit: () => JSX.Element = () => {
   const [saveState, setSaveState] = useState<SaveState>();
   const [saveStateFetched, setSaveStateFetched] = useState<boolean>(false); // needed because saveState can be empty
   const [devPageSets, setDevPageSets] = useState<any>(undefined);
+  const [hasParent, setHasParent] = useState<boolean>(true);
```

Comment:
Can we make this do the opposite and start false and then set to true if there are specific ancestors? The way you have it now is sort of the opposite of how you check things and also requires you to have the additional window check to render the loading component.

Reply:
the issue i ran into starting with false is that it redirects to the 404 page immediately, even with the null window check (basically to do nothing if window == null)